### PR TITLE
Added support for parameterized queries and mutations.

### PR DESCRIPTION
**Pull request by dblock**
On top of #19, added support for parameterized queries and mutations.

Closes https://github.com/ashkan18/graphlient/issues/1.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,9 +71,58 @@ A successful response object always contains data which can be iterated upon. Th
...
+
+### Executing Parameterized Queries and Mutations
+
+Graphlient can execute parameterized queries and mutations by providing variables as query parameters.
+
+The following query accepts an array of IDs.
+
+```ruby
+client.query(ids: [42]) do
+  query(:$ids => :'[Int]') do
```
Comment (ashkan18) [inline README.md]: oo interesting approach, I wonder if we can make this slightly more DSLy by doing things like:

``` 
client.query(ids: [42]) do
  query(:ids => [:int]) do
   ....
   end
end
```
This means on our end we need to detect query variables and add `$` to them when generating query. What do you think?

**Comment by dblock** [comment]
Two reasons for doing a parameterized query ala version 1 above:

* It passes variables separately in a different field during POST, so you don't risk injection. This is now supported by this PR.
* It allows you to pre-parse the query avoiding to have you parse it every single time. This is yet something we have to support in Graphlient, https://github.com/ashkan18/graphlient/issues/22.

File: README.md (hunk 1 above)
Comment (dblock) [reply README.md]: Good idea, opened https://github.com/ashkan18/graphlient/issues/23 for a future improvement.

**Comment by dblock** [comment]
I rebased it, it's ready to merge again.

**Pull request merged**